parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -149,9 +149,25 @@ public static Metrics footerMetrics(ParquetMetadata metadata, MetricsConfig metr
     }
 
     return new Metrics(rowCount, columnSizes, valueCounts, nullValueCounts,
+        getNanValueCounts(fieldMetrics, metricsConfig, inputSchema),
         toBufferMap(fileSchema, lowerBounds), toBufferMap(fileSchema, upperBounds));
   }
 
+  private static Map<Integer, Long> getNanValueCounts(
+      Stream<FieldMetrics> fieldMetrics, MetricsConfig metricsConfig, Schema inputSchema) {
+    if (fieldMetrics == null || inputSchema == null) {
+      return Maps.newHashMap();
+    }
+
+    return fieldMetrics
+        .filter(metrics -> {
+          String alias = inputSchema.idToAlias(metrics.getId());
+          MetricsMode metricsMode = metricsConfig.columnMode(alias);
+          return metricsMode != MetricsModes.None.get();
```

Comment:
Based on the behavior of footer metrics, it is not clear to me if we should have this metrics for `Counts` mode.

Reply:
Yes, we counts mode should include NaN counts. Null counts are passed to `increment` before the mode check above.